tests/conftest.py

```diff
@@ -127,11 +137,53 @@ def client(request):
             # ClearSession locks the device. We only do that if the PIN is set.
             client.clear_session()
 
-    client.open()
-    yield client
+    if run_ui_tests:
+        with ui_tests.screen_recording(client, request):
+            yield client
+    else:
+        yield client
+
     client.close()
 
 
+def pytest_sessionstart(session):
+    if session.config.getoption("ui") == "test":
+        report.clear_dir()
+
+
+def pytest_sessionfinish(session, exitstatus):
+    if session.config.getoption("ui") != "test":
+        return
+
+    reporter = session.config.pluginmanager.get_plugin("terminalreporter")
+    # intentionally set(), because there are multiple stages for one test in the TestReport items
+    test_names = {"passed": set(), "failed": set()}
+    for status, test in reporter.stats.items():
+        if status in ("deselected", "warnings"):
+            continue
+        if status in ("passed", "failed"):
+            # iterate through the stages to get the test name
+            for t in test:
+                test_names[status].add(get_test_name(t.nodeid))
+
+    report.index(sorted(list(test_names["passed"])), sorted(list(test_names["failed"])))
```

Comment:
istm you don't need `list()` here, and the `sorted()` should be moved either to `report.index` or maybe even inside `_report_links`